src/scipp/core/variable.py

```diff
@@ -336,7 +356,19 @@ def array(*,
     :param unit: Optional, data unit. Default=dimensionless
     :param dtype: Optional, type of underlying data. Default=None,
       in which case type is inferred from value input.
-    """
+    """  # noqa #501
+    if len(kwargs) > 1:
+        raise ValueError("Multiple unknown keyword arguments.")
+    if len(kwargs) == 1:
+        key = next(iter(kwargs))
+        if dims is not None:
+            raise ValueError(f"Given dims={dims} but dimension name already defined"
```

Comment:
`pybind11` raises `TypeError` when there is no matching overload. That is similar to the situation here. Also, regular Python functions raise `TypeError` when you call them with a wrong argument name.
I think we should also raise that here. But I am not entirely sure.